src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -185,21 +188,58 @@ public TripMonitorNotification checkTripForNewAlerts() {
         return notification;
     }
 
-    public TripMonitorNotification checkTripForDepartureDelay() {
-        long departureDelayInMinutes = TimeUnit.SECONDS.toMinutes(matchingItinerary.legs.get(0).departureDelay);
-        // First leg departure time should not exceed variance allowed.
-        if (departureDelayInMinutes >= trip.departureVarianceMinutesThreshold) {
-            return TripMonitorNotification.createDelayNotification(departureDelayInMinutes, trip.departureVarianceMinutesThreshold, DEPARTURE_DELAY);
+    /**
+     * Checks whether the trip is beginning or ending at a time greater than the allowable variance. This will check
+     * whether the departure or arrival time of the whole journey has deviated to the point where the absolute value of
+     * the variance has changed more than the trip's variance. If it has, a notification is generated.
```

Comment:
I find this javadoc to be a bit hard to follow. Suggestion:

> Check whether the arrives or departs on time. If there is a delay for the trip that exceeds the allowable variance (in either direction -- early or late), a notification is generated.

I'm not sure if that exactly captures what it currently does because it's kind of hard for me to interpret the second sentence 😅

Reply:
Examples added in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14

Reply:
Thanks for the examples, but I'm still finding the main javadoc very difficult to comprehend.